idea/src/org/jetbrains/kotlin/idea/inspections/ControlFlowWithEmptyBodyInspection.kt

```diff
@@ -21,11 +21,29 @@ class ControlFlowWithEmptyBodyInspection : AbstractKotlinInspection() {
         override fun visitIfExpression(expression: KtIfExpression) {
             val then = expression.then
             val elseKeyword = expression.elseKeyword
-            if (then.isEmptyBodyOrNull() && (elseKeyword == null || then?.hasComments() != true)) {
-                holder.registerProblem(expression, expression.ifKeyword)
-            }
-            if (elseKeyword != null && expression.`else`.isEmptyBodyOrNull()) {
-                holder.registerProblem(expression, elseKeyword)
+
+            val thenEmpty = then?.isEmptyBody()
+            val elseEmpty = expression.`else`?.isEmptyBody()
+
+            if (thenEmpty != false) {
+                // if () { }
```

Comment:
This could be read as just a commented code - please wrap it in quotes like
```
`if () { }`
```
and/or add couple meaningful words (e.g. `case ...`)